Frontend/license status updates

jsandoval81 opened the pull request:
### Requirements List
- `yarn install --ignore-engines`

### Description List
- Updated License model to account for new backend props:
   - `licenseStatus`
       - _same as previous License `status` field; only on License responses; Privilege responses still have `status`_
   - `licenseStatusName`
   - `compactEligibility`
- Updated License model tests for new fields + some clean up some gaps & consistency items
- Updated the Licensee model to move to new server-calculated eligibility status
- Updated License card UI to match updated designs
- Updated Privilege card UI to match updated designs
- Updated the purchase multi-license-select step 1 to be based off of eligible licenses
- Updated the mock data to add another licensee license so we still have all scenarios available
- Fixed minor type related to active military affiliations
- Updated transitive dependency `http-proxy-middleware` #760 

### Testing List
- `yarn test:unit:all` should run without errors or warnings
- `yarn serve` should run without errors or warnings
- `yarn build` should run without errors or warnings
- Code review
- Testing
    - _Use both mock API and live API (mock API makes it easy to test all cases)_
    - Active License cards match the updated designs
    - Inactive License cards match the updated designs
    - Active Privilege cards match the updated designs
    - Inactive Privilege cards match the updated designs

Closes #682 
Closes #760 


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
  - Added a new LicenseHome icon component with SVG rendering for use in license displays.
  - License cards now display eligibility status with visual indicators and tooltips for status descriptions.
  - License and privilege cards feature updated layouts, icons, and clearer status/eligibility displays.

- **Enhancements**
  - Improved styling for license and privilege cards, including new color schemes, typography, and responsive design adjustments.
  - Home state and eligibility information are now passed and displayed more consistently across dashboard and detail views.
  - Tooltip support added for overflowing status descriptions.

- **Localization**
  - Added English and Spanish translations for "Compact Eligible" and "Not Compact Eligible" status labels.

- **Bug Fixes**
  - Corrected typos in method names affecting military affiliation logic.

- **Tests**
  - Expanded and updated test coverage for new eligibility and status features in license and licensee models, and added tests for the new LicenseHome icon component.

- **Chores**
  - Updated mock data to include detailed license status and eligibility fields.
  - Adjusted font weight variable for extra bold text.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

jsandoval81 commented on webroot/src/components/LicenseCard/LicenseCard.ts:
```diff
@@ -11,19 +11,33 @@ import {
+        this.addStatusDescriptionExpansion();
+    }
+
     //
```
Removed event handler on unmount.

ChiefStief commented:
1) Im not sure we want to shrink the cards to fit the expiration notice. I think Id rather make the white space bigger and keep the cards the same height. It causes an odd interface with the homestate block being taller than the cards. Im open to arguments that this is better however.

2) For inactive licenses, the divider line between compact eligibility and the rest of the info is not visible to me (maybe just very low contrast

<img width="2560" alt="Screenshot 2025-04-28 at 3 29 45 PM (2)" src="https://github.com/user-attachments/assets/14e002e5-3ded-4f64-babb-64320cc03828" />

ChiefStief commented:
We lost a touch on the code coverage here fyi. If it's too hard to address no worries

ChiefStief reviewed (changes requested):
Good work, see comments

jsandoval81 commented:
@ChiefStief 
1. > **Currently, we are allowing them to choose a compact ineligible license to purchase privileges in the drop down. Is this intentional?**
    - We spoke about this offline and implemented as discussed
1. > **I prefer not to grow the home state block if a license is expired. It causes an odd interface with the homestate block being taller than the cards. Im open to arguments that this is better however.**
    - I spoke with design about this offline. They agreed to remove the expiration text under the cards, since the new UI has 1) the expiry date clearly in the card, 2) improved visual indication if the cards are inactive, and 3) the eligibility status clearly marked on each license.
1. > **For inactive licenses, the divider line between compact eligibility and the rest of the info is not visible to me (maybe just very low contrast**
    - Fixed
1. > **We lost a touch on the code coverage here fyi. If it's too hard to address no worries**
    - We did lose a few hundredths of a percent - but also there was a lot of cleanup & consolidation of some tests here that I feel improved the tests overall. For this PR, I probably won't go chasing coverage stats since we're still solidly above the requirement 🙂

ChiefStief commented:
Good with the updates assuming the following is true:
1) If the license has expired, the compact eligibility will be returned as ineligible by the backend.
2) A non home state license can never be deemed compact eligible by the backend.

Because we are now totally relying on the value returned in the `compactEligibility`  these need to be true to maintain expected functionality.

jsandoval81 commented:
@jlkravitz This is ready for your review.

landonshumway-ia commented:
> Good with the updates assuming the following is true:
> 
> 1. If the license has expired, the compact eligibility will be returned as ineligible by the backend.
> 2. A non home state license can never be deemed compact eligible by the backend.
> 
> Because we are now totally relying on the value returned in the `compactEligibility` these need to be true to maintain expected functionality.

@ChiefStief @jsandoval81 assumption 2 is not true, nor can we enforce this `compactEligibility` check, this is uploaded by the states themselves, and the backend should not modify this, so if a jurisdiction uploads a license from a non-home state then it is very possible that the non-home state license will show up as compact eligible. We can discuss this in standup tomorrow, but the frontend should not assume that just because a license is marked as `compactEligible` it means the licensee can purchase privileges for it.

jlkravitz commented on webroot/src/models/License/License.model.ts:
```diff
@@ -197,8 +211,10 @@ export class LicenseSerializer {
             history: [] as Array<LicenseHistoryItem>,
+            status: json.licenseStatus || json.status,
+            statusDescription: json.licenseStatusName,
+            eligibility: (json.type?.startsWith('license')) ? json.compactEligibility : EligibilityStatus.NA,
```
Is there a reason to test `startsWith` vs. an explicit check against the full name of the type?

jsandoval81 replied:
Licenses can be `license-home` for the home state, or just `license` for non-home state; for which `eligibility` would apply. Privileges would be the fallback eligibility value.

jlkravitz replied:
Right, sorry- I guess I wonder why we wouldn't just explicitly check that `json.type === 'license' || json.type === 'license-home'`, unless we are explicitly trying to design for the scenario where new types prefixed with license are incorporated into the system?

jsandoval81 replied:
Gotcha. I just updated that.

jlkravitz reviewed (commented):
one small comment!

jsandoval81 commented:
@jlkravitz 

> were you able to run a WCAG 2.1 AA accessibility test on the provider home page?

We were. aXe is running clean on those pages during our testing. During this feature we had quite a bit of working with the design team to make sure each updated element is sufficient contrast, etc.

This PR should be ready for your re-review.

jlkravitz reviewed (approved):
@isabeleliassen Good to merge.
